x-pack/plugins/rule_registry/server/rule_data_client/rule_data_client.ts

```diff
@@ -66,6 +68,14 @@ export class RuleDataClient implements IRuleDataClient {
     this._isWriteEnabled = isEnabled;
   }
 
+  public get lastWriterBulkError(): Error | undefined {
+    return this._lastWriterBulkError;
+  }
+
+  private set lastWriterBulkError(error: Error | undefined) {
+    this._lastWriterBulkError = error;
+  }
+
```

Comment:
Since the RuleDataClients are shared by multiple rules that can run concurrently, keeping state here exposes us to race conditions and could result in errors appearing with the wrong rules.

[This PR](https://github.com/elastic/kibana/pull/134280) by Response Ops is addressing the same issue at the framework level by modifying the `getWriter` function so it throws if writing is disabled or initialization fails. The result should be the same thing we're attempting to implement here, which is a failure status appearing in the rule details page if initialization fails or writing is disabled. Let's coordinate with that PR and ensure that we're getting the expected behavior.

Reply:
This is true, thank you for the catch! Mentioned PR looks good to me and we can reject my changes in favour of it.

There is a [comment](https://github.com/elastic/kibana/pull/134280#issuecomment-1161578496) that I left there that we might want to discuss/address before merging those changes though.

cc @ymao1